packages/runtime/src/next/client.ts

```diff
@@ -697,6 +708,22 @@ export class Makeswift {
     return { document, cacheData, key }
   }
 
+  private async requestFallbackComponentSnapshotIfNeeded(
+    initialResponse: Response,
+    id: string,
+    {
+      siteVersion,
+      locale,
+      allowLocaleFallback = true,
+    }: { siteVersion: MakeswiftSiteVersion; locale?: string; allowLocaleFallback?: boolean },
+  ): Promise<Response> {
+    const baseLocaleWasRequested = locale == null
+    if (initialResponse.status !== 404 || !allowLocaleFallback || baseLocaleWasRequested) {
+      return initialResponse
+    }
+    return await this.fetch(`v1/element-trees/${id}`, siteVersion)
+  }
+
```

Comment:
This is some very weird level of indirection that I don't think we should have. It makes it so that the calling function doesn't know whether the fallback happened or not. This is important for us to construct the appropriate snapshot and handle what the builder should show. The snapshot should include information about the originally requested resource, and this is lost if we simply return the response from the fallback.

I would strongly recommend working on this end to end, or at least doing a spike first to understand what all the moving parts will be to avoid negative work.